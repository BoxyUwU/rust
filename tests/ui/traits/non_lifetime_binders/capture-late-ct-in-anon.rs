#![feature(non_lifetime_binders, generic_const_exprs)]
//~^ WARN the feature `non_lifetime_binders` is incomplete
//~| WARN the feature `generic_const_exprs` is incomplete

fn a()
where
    for<const C: usize> [(); C + 1]: Copy,
    //~^ ERROR cannot capture late-bound const parameter in a constant
{    
}

fn b()
where
    for<const C: usize> [(); C]: Copy,
{
}

fn main() {}
