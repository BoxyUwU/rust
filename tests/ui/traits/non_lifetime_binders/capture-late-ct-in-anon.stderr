warning: the feature `non_lifetime_binders` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/capture-late-ct-in-anon.rs:1:12
   |
LL | #![feature(non_lifetime_binders, generic_const_exprs)]
   |            ^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #108185 <https://github.com/rust-lang/rust/issues/108185> for more information
   = note: `#[warn(incomplete_features)]` on by default

warning: the feature `generic_const_exprs` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/capture-late-ct-in-anon.rs:1:34
   |
LL | #![feature(non_lifetime_binders, generic_const_exprs)]
   |                                  ^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #76560 <https://github.com/rust-lang/rust/issues/76560> for more information

error: cannot capture late-bound const parameter in a constant
  --> $DIR/capture-late-ct-in-anon.rs:7:30
   |
LL |     for<const C: usize> [(); C + 1]: Copy,
   |         --------------       ^
   |         |
   |         parameter defined here

error: aborting due to previous error; 2 warnings emitted

